src/shell/input/file.c

```diff
@@ -0,0 +1,206 @@
+/************************************************************\
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* file input handling
+ *
+ * For input type "file", read a single input file on shell rank 0
+ * and place data in guest.input.
+ *
+ */
+#define FLUX_SHELL_PLUGIN_NAME "input.file"
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <assert.h>
+#include <flux/core.h>
+
+#include "src/common/libioencode/ioencode.h"
+#include "ccan/str/str.h"
+
+#include "task.h"
+#include "internal.h"
+#include "builtins.h"
+#include "input/util.h"
+
+struct file_input {
+    flux_shell_t *shell;
+    const char *path;
+    int fd;
+    flux_watcher_t *w;
+    char *ranks;
```

Comment:
Seems like ranks could be made a fixed length array and strdup/asprintf could be skipped later on?

Reply:
Yeah. However, I was trying to change as little as possible from the original plugin (as much as that was possible anyway). This code goes away in a follow on PR so I'd hate to spend time fixing it then removing it if that's ok.